### DISCUSSION TITLE
Electricity prices (forecast) 48h?

### DISCUSSION THREAD

**Discussion by akrauss** [Akkudoktor-EOS/EOS#186] [General]
I use Tibber and receive the exchange prices for the current day via their API and in the early afternoon the prices for the next day are also available in the API. 

In the best case this is over 30h in the worst case just 12h in advance. Both cases are far from the 48 values required.

Which values do you use? Do I get forecasts for the market electricity price somewhere?

I am currently using a workaround and filling in the missing values with the values of the current day. It's not pretty.

Best regards
Andreas

**Comment by eckelj** [comment]
Here is a nice overview of how to get data from different sources.
https://github.com/mampfes/ha_epex_spot/tree/main/custom_components/epex_spot
We could also use the EPEX API. But must not forget that the local grid prices might vary and push the costs. 
Grid prices will also become flexible and could be part of another optimization module or part of this price here.

**Reply by drbacke** [reply]
Yes, I am currently working on the topic and trying to generate a long-term forecast. This does not have to be 100% accurate in the end, a rough estimate is enough.  The data is not a problem. Awattar and OpenMeteo/DWD provide very good information.
